src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -69,31 +75,36 @@ private JSSynchronizationContext (Thread mainThread)
             //  fire a callback that will schedule a background job to pump the queue on the main thread.
             var awaiter = vt.AsTask().ConfigureAwait(false).GetAwaiter();
             // UnsafeOnCompleted avoids spending time flowing the execution context (we don't need it.)
-            awaiter.UnsafeOnCompleted(_DataIsAvailable);
+            awaiter.UnsafeOnCompleted(DataIsAvailable);
```

Comment:
So the problem with this change is that previously `_DataIsAvailable` was a single delegate that was allocated once when the sync context was created.  Now a new delegate object is created every time that `AwaitNewData` is called.  Generally the framework tries to avoid these kinds of allocations and reuses delegates.